service/history/workflow/mutable_state_impl.go

```diff
@@ -4503,11 +4601,13 @@ func (ms *MutableStateImpl) updatePendingEventIDs(
 		if activityInfo, ok := ms.GetActivityInfo(scheduledEventID); ok {
 			activityInfo.StartedEventId = startedEventID
 			ms.updateActivityInfos[activityInfo.ScheduledEventId] = activityInfo
+			ms.approximateSize += activityInfo.Size()
```

Comment:
compute the delta 
or probably since it's only updating one field, the size won't change. plz double check.